src/main/java/ca/fxco/configurablepistons/blocks/pistons/basePiston/BasicMovingBlockEntity.java

```diff
@@ -35,34 +35,27 @@
 
 public class BasicMovingBlockEntity extends PistonMovingBlockEntity {
 
-    /*
-     * This class overrides all the non-static methods of PistonBlockEntity
-     */
-
     protected final BasicMovingBlock MOVING_BLOCK;
 
+    /** This is only used to register the moving block entities, where none of the values are required **/
```

Comment:
I'm pretty sure this will break loading block entity data from disk. It will leave any custom block entities with the wrong block entity type and without a moving block. The basic moving block entity should have basic constructor with the type and moving block parameters, so that each custom moving block entity can pass in their block entity type and default moving block.

Reply:
I want to split the moving block out of the there. Would be nice if an inherited moving block would not need to make an inherited moving block entity

Reply:
I've made changes, both the moving block and block entity type are now set correctly